subgraphs/hop-protocol/protocols/hop-protocol/src/mainnet/bridge.ts

```diff
@@ -0,0 +1,255 @@
+  if (event.params.chainId.toString() == "42170") return;
+
+  if (NetworkConfigs.getBridgeList().includes(event.address.toHexString())) {
+    const inputToken = NetworkConfigs.getTokenAddressFromBridgeAddress(
+      event.address.toHexString()
+    )[0];
+
+    log.warning("inputToken1: {}, bridgeAddress: {}, chainId: {}", [
+      inputToken,
+      event.address.toHexString(),
+      event.params.chainId.toHexString(),
+    ]);
+
+    const poolAddress = NetworkConfigs.getPoolAddressFromChainId(
+      event.params.chainId.toString(),
+      event.address.toHexString()
+    );
+    log.warning("poolAddress2: {}, inputToken: {}, chainId: {}", [
+      poolAddress,
+      inputToken,
+      event.params.chainId.toString(),
+    ]);
+
+    const poolConfig = NetworkConfigs.getPoolDetails(poolAddress);
+
+    log.warning("poolAddress3: {}, inputToken: {}", [poolAddress, inputToken]);
+
+    const poolName = poolConfig[0];
+    const poolSymbol = poolConfig[1];
+
+    log.warning("poolAddress4: {}, inputToken: {}", [poolAddress, inputToken]);
+
+    const sdk = SDK.initializeFromEvent(
+      conf,
+      new Pricer(),
+      new TokenInit(),
+      event
+    );
+
+    log.warning("Receipient1: {}, inputToken: {}", [
+      event.params.recipient.toHexString(),
+      inputToken,
+    ]);
+    const acc = sdk.Accounts.loadAccount(event.params.recipient);
+
+    log.warning("Receipient2: {}, poolAddress: {}", [
+      event.params.recipient.toHexString(),
+      poolAddress,
+    ]);
+
+    const pool = sdk.Pools.loadPool<string>(Address.fromString(poolAddress));
+
+    log.warning("Receipient3: {}, inputToken: {}", [
+      event.params.recipient.toHexString(),
+      inputToken,
+    ]);
+
+    const token = sdk.Tokens.getOrCreateToken(Address.fromString(inputToken));
+    log.warning("Receipient4: {}, inputToken: {}", [
+      event.params.recipient.toHexString(),
+      inputToken,
+    ]);
+
+    if (!pool.isInitialized) {
+      pool.initialize(poolName, poolSymbol, BridgePoolType.LIQUIDITY, token);
+    }
+    const crossToken = sdk.Tokens.getOrCreateCrosschainToken(
+      event.params.chainId,
+      Address.fromString(
+        NetworkConfigs.getCrossTokenAddress(
+          event.params.chainId.toString(),
+          inputToken
+        )
+      ),
+      CrosschainTokenType.CANONICAL,
+      Address.fromString(inputToken)
+    );
+    log.warning("S5 - chainID: {}, inputToken: {}", [
+      event.params.chainId.toHexString(),
+      inputToken,
+    ]);
+    pool.addDestinationToken(crossToken);
+
+    acc.transferOut(
+      pool,
+      pool.getDestinationTokenRoute(crossToken)!,
+      event.params.recipient,
+      event.params.amount
+    );
+    log.warning("S6 - chainID: {}, inputToken: {}", [
+      event.params.chainId.toString(),
+      inputToken,
+    ]);
+
+    pool.addInputTokenBalance(event.params.amount);
+
+    let receipt = event.receipt;
+
+    if (receipt) {
```

Comment:
Prefer early returns over big `if` blocks :P

Reply:
Okay, I'll change that.

Reply:
let's also log a warning there saying there is no receipt as that would be an unexpected event

Reply:
Okay